src/urllib3/response.py

```diff
@@ -767,6 +779,10 @@ def _fp_read(self, amt: int | None = None) -> bytes:
           * CPython < 3.10 only when `amt` does not fit 32-bit int.
         """
         assert self._fp
+        max_chunk_amt = 2**28
+        if one_read:
+            amt = max_chunk_amt if amt is None else min(amt, max_chunk_amt)
+            return self._fp.read1(amt)
```

Comment:
It's not obvious why you have to choose a default amount and limit it to `2**28`.

I'd expect:
* something like this to be used normally:
https://github.com/urllib3/urllib3/blob/db7d96b7cc2ddc184e459301956ff203ccd91ec6/src/urllib3/response.py#L811-L812

* and only if this condition is true, the amount to be limited to `c_int_max`
https://github.com/urllib3/urllib3/blob/db7d96b7cc2ddc184e459301956ff203ccd91ec6/src/urllib3/response.py#L787-L790

Reply:
I just wanted to do something relatively quick to code initially because I wasn't sure how likely this change was to be accepted...

The reference to `StringIO` seems weird to me as well; the types indicate that the method returns bytes, but that's not what you'd get from a `StringIO`.  I'm also wondering whether that check should be moved out into a top-method somewhere so the check wouldn't need to be done on every read.